setuptools/config/setupcfg.py

```diff
@@ -174,6 +175,41 @@ def parse_configuration(
     return meta, options
 
 
+def warn_about_potential_extras_misconfig(section_options, parsed):
```

Comment:
If we are verifying the `extras_require`, does it also make sense to reuse the same logic to verify `install_requires`? (I suppose the same problems can be found there, right?)

Reply:
In that case, I'd name the function like `warn_accidental_env_marker_misconfig` because you basically want to add an env marker and it ends up parsed as a package. Then we can use the same function when parsing extras_require as well as install_requires. What do you think?

Reply:
Sounds very good!